feat(): add if-validator

develite98 opened the pull request:
Hi, `ifValidator` or `ifAsyncValidator` are simple utility functions for help to simplify change dynamically validation of Angular Reactive Form.

nartc commented on docs/src/content/docs/utilities/if-validator.md:
```diff
@@ -0,0 +1,44 @@
+
+## Usage
+
+`ifValidator` accepts a callback condition and validatorFn or validatorFn[].
+
```
```suggestion
`ifValidator` accepts a callback condition and `ValidatorFn` or `ValidatorFn[]`.
```

nartc commented on libs/ngxtension/if-validator/project.json:
```diff
@@ -0,0 +1,32 @@
+			"executor": "@nx/jest:jest",
+			"outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
+			"options": {
+				"jestConfig": "libs/ngxtension/jest.config.ts",
+				"passWithNoTests": true
```
```suggestion
				"jestConfig": "libs/ngxtension/jest.config.ts",
				"testPathPattern": ["if-validator"],
				"passWithNoTests": true
```

nartc commented on libs/ngxtension/if-validator/src/if-validator.ts:
```diff
@@ -0,0 +1,48 @@
+			}
+
+			return null;
+		} else {
```
nit: unnecessary `else`

nartc reviewed (changes requested):
A couple of nits.

Can you add some tests (either unit tests or e2e tests using the `test-app`)?

develite98 commented:
> A couple of nits.
> 
> Can you add some tests (either unit tests or e2e tests using the `test-app`)?

Hello, @nartc I've already fixed comments and added e2e testing, but as I'm not experienced with it (e2e), could you help me check it?

nartc commented on apps/test-app/src/app/if-validator/if-validator.component.ts:
```diff
@@ -0,0 +1,35 @@
+import { ifValidator } from 'ngxtension/if-validator';
+
+@Component({
+	selector: 'my-app',
```
nit: you don't need `selector` for routed component.

nartc commented on apps/test-app/src/app/if-validator/if-validator.component.ts:
```diff
@@ -0,0 +1,35 @@
+		<button (click)="changeCondition()">Change Form Condition</button>
+	`,
+})
+export default class App {
```
```suggestion
export default class App {
export default class IfValidator {
```

nartc reviewed (commented):
Test looks good enough

nartc commented:
@all-contributors please add @develite98 for code

nartc commented:
merge #45
